### PR TITLE
BAEL-8305 Map iterable to object containing iterable

### PR DESCRIPTION
**Pull request by apelan**
Examples how to map iterable to object containing iterable

### CODE REVIEW

File: mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.mapstruct.factory.Mappers;
+
+import com.baeldung.mapstruct.iterable.model.Department;
+import com.baeldung.mapstruct.iterable.model.Employee;
+
+@Mapper
+public interface DepartmentMapper {
+    DepartmentMapper INSTANCE = Mappers.getMapper(DepartmentMapper.class);
+
+    // This attempt of mapping causes error, preventing code from compiling
```
Comment (acerasoni) [inline mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java:16]: Can we remove this commented code if not required?

File: mapstruct-2/src/test/java/com/baeldung/mapstruct/iterable/DepartmentMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public class DepartmentMapperTest {
+
+    private final DepartmentMapper departmentMapper = DepartmentMapper.INSTANCE;
+    private final List<Employee> employees = Arrays.asList(
+        new Employee("John", "(555)1111-1234"),
+        new Employee("Jane", "(555)2222-4321")
+    );
+
+    @Test
+    public void givenListOfEmployees_whenMapDepartment_thenAssertMappingIsSuccessful() {
```
Comment (acerasoni) [inline mapstruct-2/src/test/java/com/baeldung/mapstruct/iterable/DepartmentMapperTest.java:23]: If using Junit5, we can drop the public modifiers on test methods

File: mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    default Department map(List<Employee> employees) {
+        Department department = new Department();
+        department.setEmployees(employees);
+        return department;
+    }
+
+    @Mapping(target = "employees", source = "employees")
+    @Mapping(target = "manager", source = "departmentManager")
+    Department mapWithManager(List<Employee> employees, Employee departmentManager);
+
```
Comment (acerasoni) [inline mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java:29]: Can we remove empty spaces at the end of the files?

File: mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java (hunk 1 above)
Comment (apelan) [reply mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java]: Not required, removed.

File: mapstruct-2/src/test/java/com/baeldung/mapstruct/iterable/DepartmentMapperTest.java (hunk 2 above)
Comment (apelan) [reply mapstruct-2/src/test/java/com/baeldung/mapstruct/iterable/DepartmentMapperTest.java]: Agree, dropped.

File: mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java (hunk 3 above)
Comment (apelan) [reply mapstruct-2/src/main/java/com/baeldung/mapstruct/iterable/DepartmentMapper.java]: Done

**Pull request merged**